### PR TITLE
[Minidump] Add extern template declarations for MinidumpFile::getListStream

### PR DESCRIPTION
**Pull request by fsfod**
Add extern template defs along with visibility macros to MinidumpFile::getListStream that is needed by MinidumpTest.cpp when llvm is built as shared library on windows with explicit visibility macros enabled.

This is part of the work to enable LLVM_BUILD_LLVM_DYLIB and LLVM plugins on window.

### CODE REVIEW

**Review by labath — COMMENTED** [review]
This was just me trying to reduce the size of the header. I may have been too clever about that.

The function itself is not that big, and there's no reason why it has to be in the cpp file. Would moving the function definition to the header be a better solution?

**Comment by fsfod** [comment]
Yes that could work.

File: llvm/include/llvm/Object/Minidump.h [hunk 1]
```diff
@@ -371,6 +377,14 @@ Expected<ArrayRef<T>> MinidumpFile::getDataSliceAs(ArrayRef<uint8_t> Data,
   return ArrayRef<T>(reinterpret_cast<const T *>(Slice->data()), Count);
 }
 
+// Needed by MinidumpTest.cpp
```
Comment (labath) [inline llvm/include/llvm/Object/Minidump.h:380]: Can we remove this? This is bound to get out of date over time. (In fact it's probably out of date already, as I suspect lldb needs this as well, but nobody is building it in a configuration where it matters).

**Review by labath — APPROVED** [review]
> Yes that could work.

Okay, then I'm going to leave it up to you to choose which one is better. I'm totally fine with moving the function to the header.

**Pull request merged**